### ISSUE TITLE
[Issue] syntax error in /config/www/app/Bin/speedtest

### ISSUE THREAD

**Issue by bladecell** [henrywhitaker3/Speedtest-Tracker#731]
All my speedtests are failing because of syntax error in /config/www/app/Bin/speedtest.

Log:

[services.d] starting services
[services.d] done.
/config/www/app/Bin/speedtest: line 1: syntax error: unexpected ";"

I am on Rpi 4 and I have created the docker image using [this](https://gist.github.com/henrywhitaker3/c6663036a0ae4355843c34cb84859d97) script, and with docker-compose image: henrywhitaker3/speedtest-tracker:latest-arm and dev-arm
 
I am not sure if docker image is supposed to work on RPi, or only manual install is possible.

**Comment by bladecell** [comment]
Manually replacing speedtest file downloaded from speedtest.net fixed the issue.

**Comment by vilse-se** [comment]
One file is missing found file on https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-x86_64-linux.tgz

How to unpack and put it in synology docker container.
ssh to your server using putty as admin@192.168.xxx.xxx
you will ned admin password for next command

#run commands
sudo find /volume*/docker -name "speedtest"
#it will find default named speedtest install folder and if file /docker/speedtest/www/app/Bin/speedtest is missing
#If it is missing do flowing, if not you have another problem

wget https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-x86_64-linux.tgz -O speedtest.tgz
#will download file fore x86_64-linux and name it speedtest.tgz

tar zxvf speedtest.tgz
#De compresses the file content 3 files only one is needed called "speedtest"

mv speedtest volume1/docker/speedtest/www/app/Bin/
#now move file to your install folder in mys server it volume1/docker/speedtest/www/app/Bin/ might be another path onyour system.

**Comment by ghost** [comment]
I am having this same issue, I have replaced the speedtest file and still same

**Comment by jimg4** [comment]
The container uses aarch64 so the correct file to download and replace is from this [link](https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-aarch64-linux.tgz).

**Comment by ghost** [comment]
Hi, Cheers but I still getting the same error when the file is replaced

**Comment by unk972** [comment]
Hello, I had a issue like that on my arm64 VM. 

/config/www/app/Bin/speedtest: line 1: syntax error: unexpected ";"

Any idea to fix it ? vilse-se give a solution for a Synology but that not work on raspi... 

find . -name speedtest
./var/lib/docker/volumes/c094c810da74f494758163063a7d28f5b88feb79cd35de45a3228da9ded77dc9/_data/log/speedtest
./var/lib/docker/volumes/c094c810da74f494758163063a7d28f5b88feb79cd35de45a3228da9ded77dc9/_data/www/app/Bin/speedtest
./var/lib/docker/volumes/a067c4b9595cafc5ea94ed0812914c54a15b7f12dcfdcfb2524db099053b66a5/_data/log/speedtest
./var/lib/docker/volumes/a067c4b9595cafc5ea94ed0812914c54a15b7f12dcfdcfb2524db099053b66a5/_data/www/app/Bin/speedtest
./var/lib/docker/volumes/abdfbd9d9e9a192ba5fd0c9d4b2f7c014aca06105a73d7739375a0e18fe8bbb2/_data/log/speedtest
./var/lib/docker/volumes/abdfbd9d9e9a192ba5fd0c9d4b2f7c014aca06105a73d7739375a0e18fe8bbb2/_data/www/app/Bin/speedtest
./var/lib/docker/overlay2/e95bcee6f891c8ad935f4c9ef2bae2476caaed9a11349b9b1000ede489b2807a/diff/etc/services.d/speedtest
./var/lib/docker/overlay2/21dad5f795cbdb69e9003c7d08ad86043daac3f340ec8a1851d6fbc70b2faa02/diff/etc/services.d/speedtest
./var/lib/docker/overlay2/f2f9e42572734cd3d78b037caeda4ab38a67c2a2ab93a8f1400d9c8daac6d5c7/diff/run/s6/services/speedtest
./var/lib/docker/overlay2/f2f9e42572734cd3d78b037caeda4ab38a67c2a2ab93a8f1400d9c8daac6d5c7/diff/etc/services.d/speedtest
./var/lib/docker/overlay2/f2f9e42572734cd3d78b037caeda4ab38a67c2a2ab93a8f1400d9c8daac6d5c7/diff/tmp/speedtest
./var/lib/docker/overlay2/f2f9e42572734cd3d78b037caeda4ab38a67c2a2ab93a8f1400d9c8daac6d5c7/diff/site/app/Bin/speedtest
./var/lib/docker/overlay2/82fce91c7eb4dd2c92570dd08dde05ffeba39f474eefe2b1f85fc09cc3831cae/diff/run/s6/services/speedtest
./var/lib/docker/overlay2/82fce91c7eb4dd2c92570dd08dde05ffeba39f474eefe2b1f85fc09cc3831cae/diff/etc/services.d/speedtest
./var/lib/docker/overlay2/82fce91c7eb4dd2c92570dd08dde05ffeba39f474eefe2b1f85fc09cc3831cae/diff/tmp/speedtest
./var/lib/docker/overlay2/82fce91c7eb4dd2c92570dd08dde05ffeba39f474eefe2b1f85fc09cc3831cae/diff/site/app/Bin/speedtest
./var/lib/docker/overlay2/e8f9931cafda0125afeca85e9f1780aae390afc709578fbfcd673d78f24ad0a7/diff/run/s6/services/speedtest
./var/lib/docker/overlay2/e8f9931cafda0125afeca85e9f1780aae390afc709578fbfcd673d78f24ad0a7/diff/etc/services.d/speedtest
./var/lib/docker/overlay2/e8f9931cafda0125afeca85e9f1780aae390afc709578fbfcd673d78f24ad0a7/diff/tmp/speedtest
./var/lib/docker/overlay2/e8f9931cafda0125afeca85e9f1780aae390afc709578fbfcd673d78f24ad0a7/diff/site/app/Bin/speedtest
./var/lib/docker/overlay2/03e6b92cc94c04b36c0f5a7d64c4d16df7ea8612e4e5736851d12ea616111eca/diff/etc/services.d/speedtest

**Comment by Elyeu1** [comment]
I have the same problem on my qnap with ARM-CPU. :(

**Comment by unk972** [comment]
Yes, it would be nice if someone can help us to find a solution! 

> Manually replacing speedtest file downloaded from speedtest.net fixed the issue.

Can u explain it?

**Comment by unk972** [comment]
After remplace "speedtest" file by that one https://install.speedtest.net/app/cli/ookla-speedtest-1.1.1-linux-aarch64.tgz

I have that message...

> You may only use this Speedtest software and information generated
from it for personal, non-commercial use, through a command line
interface on a personal computer. Your use of this software is subject
to the End User License Agreement, Terms of Use and Privacy Policy at
these URLs:

        https://www.speedtest.net/about/eula
        https://www.speedtest.net/about/terms
        https://www.speedtest.net/about/privacy

==============================================================================

To accept the message please run speedtest interactively or use the following: 

    speedtest --accept-license

**Comment by unk972** [comment]
?

**Comment by Elyeu1** [comment]
:(
If i replace the file i get the same message like unk... That's just frustrating...

**Comment by unk972** [comment]
And no help from the community :(
